pithos/plugins/notification_icon.py

```diff
@@ -92,20 +93,24 @@ def on_has_name_owner(bus, result):
                 # No fancy tray here
                 self.prepare_complete()
 
+        if not self.settings['data']:
+            self.settings['data'] = 'io.github.Pithos-tray'
+
         logging.info('Checking if org.kde.StatusNotifierWatcher is owned')
         self.bus.call('org.freedesktop.DBus', '/', 'org.freedesktop.DBus',
                       'NameHasOwner', GLib.Variant('(s)', ('org.kde.StatusNotifierWatcher',)),
                       GLib.VariantType('(b)'), Gio.DBusCallFlags.NONE, -1, None, on_has_name_owner)
 
     def on_enable(self):
+        self.icon_changed_handler = self.preferences_dialog.connect('icon-changed', self._on_icon_changed)
```

Comment:
Well it's 6 of 1 thing 1/2 a dozen of another. If we bind the the combo to the setting we still have to connect a changed signal to the handler so the icon will change...